src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2262,10 +2541,99 @@ bool TheoryStringsRewriter::checkEntailArith(Node a, bool strict)
       return true;
     }
     // TODO (#1180) : abstract interpretation goes here
+
+    // over approximation O/U
+
+    // O( x <op> y ) -> O( x ) <op> O( y )
+    // O( len( x ) ) -> ...
```

Comment:
Actually it should be len( x ) itself (see comment below).

If we wanted this to be a constant, LONG_MAX would not suffice.  This is a practical upper bound, but not theoretical upper bound.  This is a tricky issue:

Right now, there are a few rewrites that we use that are technically unsound but cannot occur in practice.  For example, if we write:

str.substr( S, 100000000000000000000000000, 2 )

where S is a constant string, we know this rewrites to empty string because we cannot even store more than LONG_MAX characters in our string data structure.  However, this rewrite is unsound in general since there is nothing in the theory that says S has length less than LONG_MAX.  Technically, the long term solution is to implement a "big string" class analogous to how "big integers" are represented. However, this is something that would very rarely occur in practice, since unlike integers, constructing a very big string is memory intensive.